services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -569,21 +569,37 @@ def __init__(self, *args: object, num_rows: int, row_group_byte_size: int) -> No
         self.row_group_byte_size = row_group_byte_size
 
 
-def get_parquet_file_and_size(url: str, hf_endpoint: str, hf_token: Optional[str]) -> tuple[pq.ParquetFile, int]:
+def retry_and_validate_get_parquet_file_num_examples_and_size(
```

Comment:
You can have two functions if you want: `get_num_examples_and_size` and `get_parquet_file_and_size` ?

Reply:
Yes sounds good ! Maybe even include the try/except in the main functions to only have 2 of them (unless you need the functions without try/except)

Reply:
like this? [1749d6a](https://github.com/huggingface/datasets-server/pull/2564/commits/1749d6a0da08d0b933e33b1809d725d56d458c99)
or do something else with exceptions?

Reply:
sounds good ! feel free to merge once it's green :)